spec/decorator.rb

```diff
@@ -0,0 +1,25 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```

Comment:
You have evidently put a lot of effort into this project 👍. Well done 💪. However, this class as it stands now throws an error when I tried a simple test with the following instance: 

```sh
person = Person.new(22, 'maximilianus')
 person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
 capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
 capitalizedTrimmedPerson.correct_name
```

I did some digging through your code and I gathered that your ```CapitalizeDecorator``` class seeks to capitalize the name of a person object passed to it but is not configured to receive an instance to the ```Person``` class and hence throws an error. This could possibly be fixed by making the ```CapitalizeDecorator``` class extend the ```Decorator``` class so that it has access to the constructor of the ```Decorator``` class as well as the @nameable instance. This would make the class look something like this: 

```sh
class CapitalizeDecorator < Decorator
  def correct_name
    @nameable.correct_name.capitalize
  end
end
```